### PR TITLE
Set DefaultMatrixAutoUpdate false and fix rig problem.

### PR DESCRIPTION
**Pull request by gonnavis**
Follow https://github.com/webaverse/app/pull/2015 which try to set `THREE.Object3D.DefaultMatrixAutoUpdate = false;`.

Based on @yaserOSource 's works, 
I added my version of initialization matrix codes, 
and solved the bone and model opposite direction problem.

There are still problems with jitter and non-steering. I'll keep trying to fix it.

And the performance still needs to be checked and improved, but at least it's relatively proper working now.

https://user-images.githubusercontent.com/10785634/147212100-d028a7f4-ee93-492f-9b7d-66a539674270.mp4

### CODE REVIEW

File: avatars/avatars.js [hunk 1]
```diff
@@ -1612,7 +1612,8 @@ class Avatar {
     this.legsManager.rightLeg.foot.position.copy(setups.rightFoot);
     if (setups.rightToe) this.legsManager.rightLeg.toe.position.copy(setups.rightToe);
 
-    this.shoulderTransforms.root.updateMatrixWorld();
+    this.shoulderTransforms.root.traverse(child => child.updateMatrix())
```
Comment (yaserOSource) [inline avatars/avatars.js:1615]: this is what i was missing

**Comment by yaserOSource** [comment]
node:internal/process/esm_loader:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\GitHub\gonnavis2\app\node_modules\wsrtc\wsrtc-server.mjs' imported from D:\GitHub\gonnavis2\app\index.mjs
    at new NodeError (node:internal/errors:363:5)
    at finalizeResolution (node:internal/modules/esm/resolve:307:11)
    at moduleResolve (node:internal/modules/esm/resolve:742:10)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:853:11)
    at Loader.resolve (node:internal/modules/esm/loader:89:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:73:40)
    at link (node:internal/modules/esm/module_job:72:36) {
  code: 'ERR_MODULE_NOT_FOUND'
  
  get this error on testing this branch / repo

**Comment by gonnavis** [comment]
Fixed can't steer/rotation problem.

https://user-images.githubusercontent.com/10785634/147235414-4167b77e-23bb-4954-80c4-71d38b6c2824.mp4

Still glitch, keep fixing.

File: avatars/avatars.js [hunk 2]
```diff
@@ -887,8 +850,12 @@ class Avatar {
     }
 
 
-    const _getOffset = (bone, parent = bone?.parent) => bone && bone.getWorldPosition(new THREE.Vector3()).sub(parent.getWorldPosition(new THREE.Vector3()));
-
+    const _getOffset = ((bone, parent = bone?.parent) => {
+      bone && bone.updateMatrix();
+      return bone && bone.getWorldPosition(new THREE.Vector3()).sub(parent.getWorldPosition(new THREE.Vector3()));
```
Comment (avaer) [inline avatars/avatars.js:855]: Can we use a local and not allocate?

**Comment by gonnavis** [comment]
The glitch turned out to be that I changed the monitor's frame rate from 60 to 300 yesterday to test performance.
So it's not related to this pr, a new PR compatible with high frame rate is needed later.
PS: Tested that only walk and run animations affected by fps, jump and fly always correct.

**Comment by gonnavis** [comment]
> The glitch turned out to be that I changed the monitor's frame rate from 60 to 300 yesterday to test performance.
> So it's not related to this pr, a new pr compatible with high frame rate is needed later.

`master` branch has not glitch at all frame rate, so the glitch still related to this pr, need to be fixed.

**Comment by gonnavis** [comment]
We have reached a consensus that not to set "THREE.Object3D.DefaultMatrixAutoUpdate = false;" first.

**Pull request closed without merging**